keras/backend/theano_backend.py

```diff
@@ -1551,7 +1551,9 @@ def relu(x, alpha=0., max_value=None):
     return x
 
 
-def softmax(x):
+def softmax(x, axis=-1):
+    if axis != -1:
```

Comment:
In general we can use the default implementation if `axis == 1 or axis == x.ndim - 1`